jib-core/src/main/java/com/google/cloud/tools/jib/frontend/MainClassFinder.java

```diff
@@ -36,20 +37,49 @@
   /** Helper for loading a .class file. */
   private static class ClassFileLoader extends ClassLoader {
 
-    private final Path classFile;
+    private Path rootDirectory;
+    private Path classFile;
 
-    private ClassFileLoader(Path classFile) {
+    private ClassFileLoader(Path classFile, Path rootDirectory) {
       this.classFile = classFile;
+      this.rootDirectory = rootDirectory;
     }
 
     @Nullable
     @Override
     public Class findClass(@Nullable String name) {
       try {
+        // Name is only ever null when we call findClass manually, and not null otherwise. If null,
+        // we should resolve the correct filename.
+        if (name != null) {
+          classFile = rootDirectory;
+          List<String> folders = Splitter.on(".").splitToList(name);
+          for (int index = 0; index < folders.size() - 1; index++) {
```

Comment:
Should use iterator style rather than indexing. (like first remove the last element from folders)

Reply:
`splitToList` returns an unmodifiable list.

Reply:
Nevermind, works if I wrap it in an ArrayList